Compile/Upload failure when using some debug features in 2.4.2 ONLY

### Basic Infos

- [X] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [X] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [X] I have tested that the issue is present in current master branch (aka latest git).
- [X] I have searched the issue tracker for a similar issue.
- [ ] If there is a stack dump, I have decoded it.
- [X] I have filled out all fields below.

#### Platform

- Hardware:					[ESP-8266]
- Core Version:			[2.4.2]
- Development Env:		[Arduino IDE]
- Operating System:		[Ubuntu]

### Settings in IDE

- Module:  			[Adafruit Feather Huzzah]
- Flash Mode:			[?]
- Flash Size: 		[4MB/1MB]
- lwip Variant:   [v2 Lower Memory]
- Reset Method:		[?]
- Flash Frequency:	[40Mhz]
- CPU Frequency:		[80Mhz]
- Upload Using:		[SERIAL]
- Upload Speed:   [921600] 


### Problem Description

When using some debug systems (like Core or OOM) the sketch refuses to compile/upload, most likely due to a conflict with ESP8266mDNS in Arduino OTA.

Is there any other way debugging could be done? I included the minimal code that still reproduces the problem.

Is it just my configuration or an actual bug?

### [MCVE](https://stackoverflow.com/help/mcve) Sketch

```cpp



#include <ArduinoOTA.h>


void setup() {
}
void loop() {

}
```

### Debug Messages

```
Arduino: 1.8.5 (Linux), Board: "Adafruit Feather HUZZAH ESP8266, 80 MHz, Flash, 4M (1M SPIFFS), v2 Lower Memory, Serial, SSL+TLS_MEM+HTTP_CLIENT+HTTP_SERVER+CORE+WIFI+HTTP_UPDATE+UPDATER+OTA+OOM, Only Sketch, 921600"

/home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/arduino-builder -dump-prefs -logger=machine -hardware /home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/hardware -hardware /home/simon/.arduino15/packages -tools /home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/tools-builder -tools /home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/hardware/tools/avr -tools /home/simon/.arduino15/packages -built-in-libraries /home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/libraries -libraries /home/simon/Arduino/libraries -fqbn=esp8266:esp8266:huzzah:CpuFrequency=80,VTable=flash,FlashSize=4M1M,LwIPVariant=v2mss536,Debug=Serial,DebugLevel=SSLTLS_MEMHTTP_CLIENTHTTP_SERVERCOREWIFIHTTP_UPDATEUPDATEROTAOOM,FlashErase=none,UploadSpeed=921600 -ide-version=10805 -build-path /tmp/arduino_build_171356 -warnings=none -build-cache /tmp/arduino_cache_228890 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.xtensa-lx106-elf-gcc.path=/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2 -prefs=runtime.tools.mkspiffs.path=/home/simon/.arduino15/packages/esp8266/tools/mkspiffs/0.2.0 -prefs=runtime.tools.esptool.path=/home/simon/.arduino15/packages/esp8266/tools/esptool/0.4.13 -verbose /home/simon/Arduino/bug_report/bug_report.ino
/home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/arduino-builder -compile -logger=machine -hardware /home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/hardware -hardware /home/simon/.arduino15/packages -tools /home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/tools-builder -tools /home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/hardware/tools/avr -tools /home/simon/.arduino15/packages -built-in-libraries /home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/libraries -libraries /home/simon/Arduino/libraries -fqbn=esp8266:esp8266:huzzah:CpuFrequency=80,VTable=flash,FlashSize=4M1M,LwIPVariant=v2mss536,Debug=Serial,DebugLevel=SSLTLS_MEMHTTP_CLIENTHTTP_SERVERCOREWIFIHTTP_UPDATEUPDATEROTAOOM,FlashErase=none,UploadSpeed=921600 -ide-version=10805 -build-path /tmp/arduino_build_171356 -warnings=none -build-cache /tmp/arduino_cache_228890 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.xtensa-lx106-elf-gcc.path=/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2 -prefs=runtime.tools.mkspiffs.path=/home/simon/.arduino15/packages/esp8266/tools/mkspiffs/0.2.0 -prefs=runtime.tools.esptool.path=/home/simon/.arduino15/packages/esp8266/tools/esptool/0.4.13 -verbose /home/simon/Arduino/bug_report/bug_report.ino
Using board 'huzzah' from platform in folder: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2
Using core 'esp8266' from platform in folder: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2
Detecting libraries used...
"/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/lwip2/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include" "-I/tmp/arduino_build_171356/core" -c -w  -Os -g -mlongcalls -mtext-section-literals -fno-exceptions -fno-rtti -falign-functions=4 -std=c++11  -ffunction-sections -fdata-sections -w -x c++ -E -CC -DF_CPU=80000000L -DLWIP_OPEN_SRC -DTCP_MSS=536 -DDEBUG_ESP_PORT=Serial -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_OOM -DARDUINO=10805 -DARDUINO_ESP8266_ESP12 -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="ESP8266_ESP12"   -DESP8266 "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/variants/adafruit" "/tmp/arduino_build_171356/sketch/bug_report.ino.cpp" -o "/dev/null"
"/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/lwip2/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include" "-I/tmp/arduino_build_171356/core" -c -w  -Os -g -mlongcalls -mtext-section-literals -fno-exceptions -fno-rtti -falign-functions=4 -std=c++11  -ffunction-sections -fdata-sections -w -x c++ -E -CC -DF_CPU=80000000L -DLWIP_OPEN_SRC -DTCP_MSS=536 -DDEBUG_ESP_PORT=Serial -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_OOM -DARDUINO=10805 -DARDUINO_ESP8266_ESP12 -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="ESP8266_ESP12"   -DESP8266 "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/variants/adafruit" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ArduinoOTA" "/tmp/arduino_build_171356/sketch/bug_report.ino.cpp" -o "/dev/null"
"/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/lwip2/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include" "-I/tmp/arduino_build_171356/core" -c -w  -Os -g -mlongcalls -mtext-section-literals -fno-exceptions -fno-rtti -falign-functions=4 -std=c++11  -ffunction-sections -fdata-sections -w -x c++ -E -CC -DF_CPU=80000000L -DLWIP_OPEN_SRC -DTCP_MSS=536 -DDEBUG_ESP_PORT=Serial -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_OOM -DARDUINO=10805 -DARDUINO_ESP8266_ESP12 -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="ESP8266_ESP12"   -DESP8266 "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/variants/adafruit" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ArduinoOTA" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src" "/tmp/arduino_build_171356/sketch/bug_report.ino.cpp" -o "/dev/null"
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ArduinoOTA/ArduinoOTA.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/BearSSLHelpers.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/CertStoreBearSSL.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFi.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFiScan.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiClient.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiClientSecureAxTLS.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiServer.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiServerSecureAxTLS.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiServerSecureBearSSL.cpp
Using cached library dependencies for file: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiUdp.cpp
"/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/lwip2/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include" "-I/tmp/arduino_build_171356/core" -c -w  -Os -g -mlongcalls -mtext-section-literals -fno-exceptions -fno-rtti -falign-functions=4 -std=c++11  -ffunction-sections -fdata-sections -w -x c++ -E -CC -DF_CPU=80000000L -DLWIP_OPEN_SRC -DTCP_MSS=536 -DDEBUG_ESP_PORT=Serial -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_OOM -DARDUINO=10805 -DARDUINO_ESP8266_ESP12 -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="ESP8266_ESP12"   -DESP8266 "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/variants/adafruit" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ArduinoOTA" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS" "/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.cpp" -o "/dev/null"
Generating function prototypes...
"/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/lwip2/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include" "-I/tmp/arduino_build_171356/core" -c -w  -Os -g -mlongcalls -mtext-section-literals -fno-exceptions -fno-rtti -falign-functions=4 -std=c++11  -ffunction-sections -fdata-sections -w -x c++ -E -CC -DF_CPU=80000000L -DLWIP_OPEN_SRC -DTCP_MSS=536 -DDEBUG_ESP_PORT=Serial -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_OOM -DARDUINO=10805 -DARDUINO_ESP8266_ESP12 -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="ESP8266_ESP12"   -DESP8266 "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/variants/adafruit" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ArduinoOTA" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS" "/tmp/arduino_build_171356/sketch/bug_report.ino.cpp" -o "/tmp/arduino_build_171356/preproc/ctags_target_for_gcc_minus_e.cpp"
"/home/simon/Downloads/arduino-1.8.5-linux64/arduino-1.8.5/tools-builder/ctags/5.8-arduino11/ctags" -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives "/tmp/arduino_build_171356/preproc/ctags_target_for_gcc_minus_e.cpp"
Compiling sketch...
"/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/lwip2/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include" "-I/tmp/arduino_build_171356/core" -c -w  -Os -g -mlongcalls -mtext-section-literals -fno-exceptions -fno-rtti -falign-functions=4 -std=c++11 -MMD -ffunction-sections -fdata-sections -DF_CPU=80000000L -DLWIP_OPEN_SRC -DTCP_MSS=536 -DDEBUG_ESP_PORT=Serial -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_OOM -DARDUINO=10805 -DARDUINO_ESP8266_ESP12 -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="ESP8266_ESP12"   -DESP8266 "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/variants/adafruit" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ArduinoOTA" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS" "/tmp/arduino_build_171356/sketch/bug_report.ino.cpp" -o "/tmp/arduino_build_171356/sketch/bug_report.ino.cpp.o"
Compiling libraries...
Compiling library "ArduinoOTA"
Using previously compiled file: /tmp/arduino_build_171356/libraries/ArduinoOTA/ArduinoOTA.cpp.o
Compiling library "ESP8266WiFi"
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/BearSSLHelpers.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/CertStoreBearSSL.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/ESP8266WiFi.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/ESP8266WiFiAP.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/ESP8266WiFiGeneric.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/ESP8266WiFiMulti.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/ESP8266WiFiSTA.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/ESP8266WiFiScan.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/WiFiClient.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/WiFiClientSecureAxTLS.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/WiFiClientSecureBearSSL.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/WiFiServer.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/WiFiServerSecureAxTLS.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/WiFiServerSecureBearSSL.cpp.o
Using previously compiled file: /tmp/arduino_build_171356/libraries/ESP8266WiFi/WiFiUdp.cpp.o
Compiling library "ESP8266mDNS"
"/home/simon/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/lwip2/include" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include" "-I/tmp/arduino_build_171356/core" -c -w  -Os -g -mlongcalls -mtext-section-literals -fno-exceptions -fno-rtti -falign-functions=4 -std=c++11 -MMD -ffunction-sections -fdata-sections -DF_CPU=80000000L -DLWIP_OPEN_SRC -DTCP_MSS=536 -DDEBUG_ESP_PORT=Serial -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_OOM -DARDUINO=10805 -DARDUINO_ESP8266_ESP12 -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="ESP8266_ESP12"   -DESP8266 "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/variants/adafruit" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ArduinoOTA" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src" "-I/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS" "/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.cpp" -o "/tmp/arduino_build_171356/libraries/ESP8266mDNS/ESP8266mDNS.cpp.o"
**In file included from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266/Arduino.h:307:0,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/CertStoreBearSSL.h:23,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h:29,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiServerSecureBearSSL.h:24,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/WiFiServerSecure.h:21,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFi.h:41,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.h:46,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.cpp:41:
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266/umm_malloc/umm_malloc_cfg.h:181:40: error: mem_debug_file causes a section type conflict with __c
 #define malloc(s) ({ static const char mem_debug_file[] ICACHE_RODATA_ATTR STORE_ATTR = __FILE__; malloc_loc(s, mem_debug_file, __LINE__); })
                                        ^
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/include/mem.h:43:19: note: in expansion of macro 'malloc'
 #define os_malloc malloc
                   ^
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.cpp:761:36: note: in expansion of macro 'os_malloc'
         answer->hostname = (char *)os_malloc(strlen(answerHostName) + 1);
                                    ^
In file included from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266/WString.h:29:0,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266/IPAddress.h:24,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFi.h:31,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.h:46,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.cpp:41:
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266/pgmspace.h:16:51: note: '__c' was declared here
 #define PSTR(s) (__extension__({static const char __c[] PROGMEM = (s); &__c[0];}))
                                                   ^
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include/assert.h:18:29: note: in expansion of macro 'PSTR'
              __ASSERT_FUNC, PSTR(#__e)))
                             ^
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/include/UdpContext.h:154:9: note: in expansion of macro 'assert'
         assert(isValidOffset(pos));
         ^
Using library ArduinoOTA at version 1.0 in folder: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ArduinoOTA 
Using library ESP8266WiFi at version 1.0 in folder: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi 
Using library ES                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/ESP8266WiFi.h:31,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.h:46,
                 from /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS/ESP8266mDNS.cpp:41:
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/cores/esp8266/pgmspace.h:16:51: note: '__c' was declared here
 #define PSTR(s) (__extension__({static const char __c[] PROGMEM = (s); &__c[0];}))
                                                   ^
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/tools/sdk/libc/xtensa-lx106-elf/include/assert.h:18:29: note: in expansion of macro 'PSTR'
              __ASSERT_FUNC, PSTR(#__e)))
                             ^**
/home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266WiFi/src/include/UdpContext.h:154:9: note: in expansion of macro 'assert'
         assert(isValidOffset(pos));
         ^P8266mDNS in folder: /home/simon/.arduino15/packages/esp8266/hardware/esp8266/2.4.2/libraries/ESP8266mDNS (legacy)
exit status 1
Error compiling for board Adafruit Feather HUZZAH ESP8266.
```

--- a/cores/esp8266/Print.cpp
+++ b/cores/esp8266/Print.cpp
@@ -35,7 +35,7 @@
 size_t Print::write(const uint8_t *buffer, size_t size) {
 
 #ifdef DEBUG_ESP_CORE
-    static char not_the_best_way [] ICACHE_RODATA_ATTR STORE_ATTR = "Print::write(data,len) should be overridden for better efficiency\r\n";
+    static char not_the_best_way [] PROGMEM STORE_ATTR = "Print::write(data,len) should be overridden for better efficiency\r\n";
     static bool once = false;
     if (!once) {
         once = true;

--- a/cores/esp8266/heap.c
+++ b/cores/esp8266/heap.c
@@ -84,9 +84,9 @@ void* ICACHE_RAM_ATTR pvPortZalloc(size_t size, const char* file, int line)
 #undef calloc
 #undef realloc
 
-static const char oom_fmt[]   ICACHE_RODATA_ATTR STORE_ATTR = ":oom(%d)@?\n";
-static const char oom_fmt_1[] ICACHE_RODATA_ATTR STORE_ATTR = ":oom(%d)@";
-static const char oom_fmt_2[] ICACHE_RODATA_ATTR STORE_ATTR = ":%d\n";
+static const char oom_fmt[]   PROGMEM STORE_ATTR = ":oom(%d)@?\n";
+static const char oom_fmt_1[] PROGMEM STORE_ATTR = ":oom(%d)@";
+static const char oom_fmt_2[] PROGMEM STORE_ATTR = ":%d\n";
 
 void* malloc (size_t s)
 {

--- a/cores/esp8266/pgmspace.h
+++ b/cores/esp8266/pgmspace.h
@@ -10,7 +10,19 @@
 #include "ets_sys.h"
 #include "osapi.h"
 
-#define PROGMEM     ICACHE_RODATA_ATTR
+// Since __section__ is supposed to be only use for global variables,
+// there could be conflicts when a static/inlined function has them in the
+// same file as a non-static PROGMEM object.
+// Ref: https://gcc.gnu.org/onlinedocs/gcc-3.2/gcc/Variable-Attributes.html
+// Place each progmem object into its own named section, avoiding conflicts
+
+// The following two macros cause a parameter to be enclosed in quotes
+// by the preopressor (i.e. for concatenating ints to strings)
+#define __STRINGIZE_NX(A) #A
+#define __STRINGIZE(A) __STRINGIZE_NX(A)
+
+#define PROGMEM      __attribute__((section( ".irom.text." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__))))
+
 #define PGM_P  		const char *
 #define PGM_VOID_P  const void *
 #define PSTR(s) (__extension__({static const char __c[] PROGMEM = (s); &__c[0];}))

--- a/cores/esp8266/uart.c
+++ b/cores/esp8266/uart.c
@@ -41,12 +41,13 @@
  *
  */
 #include "Arduino.h"
+#include <pgmspace.h>
 #include "uart.h"
 #include "esp8266_peri.h"
 #include "user_interface.h"
 #include "uart_register.h"
 
-const char overrun_str [] ICACHE_RODATA_ATTR STORE_ATTR = "uart input full!\r\n";
+const char overrun_str [] PROGMEM STORE_ATTR = "uart input full!\r\n";
 static int s_uart_debug_nr = UART0;
 
 

--- a/cores/esp8266/umm_malloc/umm_malloc_cfg.h
+++ b/cores/esp8266/umm_malloc/umm_malloc_cfg.h
@@ -178,7 +178,8 @@ extern char _heap_start;
 // this must be outside from "#ifndef _UMM_MALLOC_CFG_H"
 // because Arduino.h's <cstdlib> does #undef *alloc
 // Arduino.h recall us to redefine them
-#define malloc(s) ({ static const char mem_debug_file[] ICACHE_RODATA_ATTR STORE_ATTR = __FILE__; malloc_loc(s, mem_debug_file, __LINE__); })
-#define calloc(n,s) ({ static const char mem_debug_file[] ICACHE_RODATA_ATTR STORE_ATTR = __FILE__; calloc_loc(n, s, mem_debug_file, __LINE__); })
-#define realloc(p,s) ({ static const char mem_debug_file[] ICACHE_RODATA_ATTR STORE_ATTR = __FILE__; realloc_loc(p, s, mem_debug_file, __LINE__); })
+#include <pgmspace.h>
+#define malloc(s) ({ static const char mem_debug_file[] PROGMEM STORE_ATTR = __FILE__; malloc_loc(s, mem_debug_file, __LINE__); })
+#define calloc(n,s) ({ static const char mem_debug_file[] PROGMEM STORE_ATTR = __FILE__; calloc_loc(n, s, mem_debug_file, __LINE__); })
+#define realloc(p,s) ({ static const char mem_debug_file[] PROGMEM STORE_ATTR = __FILE__; realloc_loc(p, s, mem_debug_file, __LINE__); })
 #endif /* DEBUG_ESP_OOM */

--- a/libraries/ESP8266WebServer/src/detail/mimetable.cpp
+++ b/libraries/ESP8266WebServer/src/detail/mimetable.cpp
@@ -5,7 +5,7 @@ namespace mime
 {
 
 // Table of extension->MIME strings stored in PROGMEM, needs to be global due to GCC section typing rules
-const Entry mimeTable[maxType] ICACHE_RODATA_ATTR = 
+const Entry mimeTable[maxType] PROGMEM = 
 {
     { ".html", "text/html" },
     { ".htm", "text/html" },

--- a/libraries/ESP8266WiFi/examples/WiFiHTTPSServer/make-self-signed-cert.sh
+++ b/libraries/ESP8266WiFi/examples/WiFiHTTPSServer/make-self-signed-cert.sh
@@ -4,11 +4,11 @@
 # Replace your-name-here with somethine appropriate before running and use
 # the generated .H files in your code as follows:
 #
-#      static const uint8_t rsakey[]  ICACHE_RODATA_ATTR = {
+#      static const uint8_t rsakey[] PROGMEM = {
 #        #include "key.h"
 #      };
 #
-#      static const uint8_t x509[] ICACHE_RODATA_ATTR = {
+#      static const uint8_t x509[] PROGMEM = {
 #        #include "x509.h"
 #      };
 #

